src/VectorizeLoops.cpp

```diff
@@ -10,23 +10,209 @@
 #include "IREquality.h"
 #include "ExprUsesVar.h"
 #include "Solve.h"
+#include "Simplify.h"
+#include "CSE.h"
 
 namespace Halide {
 namespace Internal {
 
 using std::string;
 using std::vector;
+using std::pair;
+
+// For a given var, replace expressions like shuffle_vector(var, 4)
+// with var.lane.4
+class ReplaceShuffleVectors : public IRMutator {
+    string var;
+
+    using IRMutator::visit;
+
+    void visit(const Call *op) {
+        const Variable *v;
+        const int64_t *i;
+        if (op->is_intrinsic(Call::shuffle_vector) &&
+            op->args.size() == 2 &&
+            (v = op->args[0].as<Variable>()) &&
```

Comment:
Edit: after making this comment I see that usage of this idiom is pretty much everywhere in this file, so... nevermind